framework/src/mesh/MooseMesh.C

```diff
@@ -632,19 +635,69 @@ MooseMesh::buildLowerDMesh()
   // remove existing lower-d element first
   std::set<Elem *> deleteable_elems;
   for (auto & elem : mesh.element_ptr_range())
-    if (elem->subdomain_id() == Moose::INTERNAL_SIDE_LOWERD_ID ||
-        elem->subdomain_id() == Moose::BOUNDARY_SIDE_LOWERD_ID)
+    if (_lower_d_interior_blocks.count(elem->subdomain_id()) ||
+        _lower_d_boundary_blocks.count(elem->subdomain_id()))
       deleteable_elems.insert(elem);
     else if (elem->n_sides() > max_n_sides)
       max_n_sides = elem->n_sides();
 
   for (auto & elem : deleteable_elems)
     mesh.delete_elem(elem);
+  for (const auto & id : _lower_d_interior_blocks)
+    _mesh_subdomains.erase(id);
+  for (const auto & id : _lower_d_boundary_blocks)
+    _mesh_subdomains.erase(id);
```

Comment:
why do that?

Reply:
This is for reentering this function safely. First call does not need this, but if we recalling this function without this, we will hit the error checks. This part of code is essentially performing clearance of what has been done by this function.

Reply:
If I define lower D blocks in my mesh generation, this will delete all of them

Reply:
No because `_lower_d_interior_blocks` only contains blocks in this name `INTERNAL_SIDE_LOWERD_SUBDOMAIN_xxx`.